pkg/chaosdaemon/httpchaos_server.go

```diff
@@ -64,15 +64,15 @@ func (s *DaemonServer) ApplyHttpChaos(ctx context.Context, in *pb.ApplyHttpChaos
 	log := log.WithValues("Request", in)
 	log.Info("applying http chaos")
 
-	if in.Instance == 0 {
+	stdio := s.backgroundProcessManager.Stdio(int(in.Instance), in.StartTime)
+	for stdio == nil {
```

Comment:
How about adding a times limit? If the instances can never be found, here will block the whole program

Reply:
+1, it would be stuck in here if `chaos-tproxy` failed to start by invalid arguments or other potential bugs.

And the deeper issue is that we do not specify the lifecycle/status of a process managed by `bpm`, we always assume that processes controlled by bpm SHOULD work well, but that's out of the scope of this PR.

PTAL @Hexilee

Reply:
@Hexilee PTAL

Reply:
> How about adding a times limit? If the instances can never be found, here will block the whole program

I think the whole program shouldn't be blocked as this is just a gRPC call, only a goroutine will be blocked. Moreover, this call will timeout if the instances can never be found.

> +1, it would be stuck in here if `chaos-tproxy` failed to start by invalid arguments or other potential bugs.
> 
> And the deeper issue is that we do not specify the lifecycle/status of a process managed by `bpm`, we always assume that processes controlled by bpm SHOULD work well, but that's out of the scope of this PR.
> 
> PTAL @Hexilee

If chaos-tproxy failed to start by invalid arguments or other potential bugs, the experiment SHOULD fail by gRPC timeout.

Reply:
The first not-found may be caused by restarting of chaos-daemon, so we try to recreate one; the second not-found is unexpected, so we should retry once only.

PTAL changes @STRRL @cwen0